### ISSUE TITLE
[Feature Request] denylist for Wi-Fi networks

### ISSUE THREAD

**Issue by y20k** [mullvad/mullvadvpn-app#2049]
## Issue report

Operating system: Mobile

App version: -

### Issue description

Some corporate networks or university networks (like the popular [eduroam](https://www.eduroam.org/)) do not allow opening VPN connections. In that setting the Mullvad mobile app is constantly trying connect. I am not 100% sure, but it seems that, as long as the app is trying to reconnect after a network switch, there is no network connection at all. 

As an example my day to day experience: I work at an university. My Android 9 phone has the Mullvad app running all the time. It is configured as an Always-On VPN - the problem arises without that setting, too. At home (Wi-Fi) and during the commute (LTE) my phone has an VPN secured internet connection. As soon as the phone connects to the university eduroam Wi-Fi, Mullvad tries to reconnect saying "blocking internet". My phone has no internet connection until I manually disable Mullvad.

### Feature request
A denylist for WiFi networks could solve the problem. When connected to networks, that have been put on the denylist, Mulvad should not try to connect.

**Comment by faern** [comment]
This is basically a duplicate of #1884 and #783. Those two tickets contain info about why this feature does not yet exist. I'm closing this now.

**Comment by faern** [comment]
Regarding your problem more specifically, and not a list of SSIDs to ignore. Maybe it would work on certain ports? Maybe eduroam only blocks some traffic that it does not understand. It might be possible that WireGuard would work over port 53? The Android app does not yet have any setting to allow changing the port. But if we know it would help some people be unblocked we might prioritize adding it.

If you want to. You can try installing the [vanilla WireGuard app](https://www.wireguard.com/install/#android-play-store-f-droid) on your phone and download a configuration for it from [our configuration generator](https://mullvad.net/account/#/wireguard-config/) and make it use port 53 (under advanced) and see if that works on your university connection?

I'll re-open this for the sake of just seeing if we can work around the censoring you are experiencing.

**Comment by y20k** [comment]
@faern 

I tried the approach you suggested using port 53. Yesterday I was unable to get the standard WireGuard app running on my phone, while I was at home(, where the Mullvad app works without issues). 

Today I tried another approach using my desktop iMac at work, which is also connected to the university network. First I installed the Mullvad App. It behaved the same as the phone. It was constantly trying to connect. Next I installed the vanilla Wireguard app through the App Store. I installed a custom profile (port 53). There was no connection possible.

Two more tidbits. I am able to connect to a Cisco-VPN (vpnc) from within the university. I am NOT able to use DNS over TLS on my phone, when it is connected to the university network.

Anyway. Thank you for your attempt to help me with my special problem. 

I still think that a denylist would be a good addition to the app. Even if we had succeeded to solve my problem, there surely will be other cases. Other networks, that have effective ways to prevent VPN use. Ideally the app had both. Custom port selection AND denylist. That latter as a last resort. 

Adding a network to the denylist would be an active step by the user. In that regard it is a security risk a user takes consciously. The problem that SSIDs can be spoofed remains a problem though. But only if a user would put a weekly authenticated network in the denylist, as far as I understand. 

Enough lobbying for this feature. I totally understand, I you want to close the issue. I will remain a happy Mullvad customer in any case. Thanks and keep up the good work.

**Comment by faern** [comment]
I'm closing this as inactive and unclear exactly what we are trying to resolve. The Wi-fi denylist thingy is covered in other issues and the fact that you can't connect is more of an anti-censorship / obfuscation issue. Please open a new issue (or look for existing ones) to discuss working around being blocked in certain networks if relevant.
